Lib/ldap/ldapobject.py

```diff
@@ -144,7 +144,10 @@ def _bytesify_input(self, arg_name, value):
         return value
       else:
         if self.bytes_mode_hardfail:
-          raise TypeError("All provided fields *must* be bytes when bytes mode is on; got %r" % (value,))
+          raise TypeError(
+            "All provided fields *must* be bytes when bytes mode is on; "
+            "got {}={!r}".format(arg_name, value)
+          )
```

Comment:
We did a similar thing with the warnings, but removed the value altogether.
Error messages might end up in logs as well; how would you feel about *only* mentioning the arg_name?